pkg/slicegwrecycler/recycler.go

```diff
@@ -0,0 +1,207 @@
+package slicegwrecycler
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	monitoringEvents "github.com/kubeslice/kubeslice-monitoring/pkg/events"
+	kubeslicev1beta1 "github.com/kubeslice/worker-operator/api/v1beta1"
+	"github.com/kubeslice/worker-operator/controllers"
+
+	ossEvents "github.com/kubeslice/worker-operator/events"
+	sidecar "github.com/kubeslice/worker-operator/pkg/gwsidecar"
+	hub "github.com/kubeslice/worker-operator/pkg/hub/hubclient"
+	"github.com/kubeslice/worker-operator/pkg/logger"
+	"github.com/kubeslice/worker-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	NumberOfGateways = 2
+)
+
+type recyclerClient struct {
+}
+
+func NewRecyclerClient() (*recyclerClient, error) {
+	return &recyclerClient{}, nil
+}
+
+func (r recyclerClient) TriggerFSM(ctx context.Context, sliceGw *kubeslicev1beta1.SliceGateway, slice *kubeslicev1beta1.Slice, hubClient *hub.HubClientConfig, meshClient client.Client, gatewayPod *corev1.Pod,
```

Comment:
I feel the we are overloading lots parameters to APIs
for instance TriggerFSM is having 9 arguments
Can we have this as a part of recyclerClient object?